src/test/common/ios/simulatorPlist.test.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {SimulatorPlist} from "../../../common/ios/simulatorPlist";
+
+import * as assert from "assert";
+import * as path from "path";
+import * as Q from "q";
+import * as sinon from "sinon";
+
+suite("plistBuddy", function() {
+    suite("commonContext", function() {
+        test("Should correctly find the NSUserDefaults plist file for the simulator", function() {
+            const projectRoot = path.join("project", "root");
+
+            const bundleId = "com.contoso.app";
+
+            const findSimulatorHomeCommand = "xcrun simctl getenv booted HOME";
+            // The emulator's home folder is /simulator/home
+            const findSimulatorHomeResult = path.join("/", "simulator", "home");
+
+            const prefix = path.join("Containers", "Data", "Application");
+            const suffix = path.join("Library", "Preferences");
+
+            // The emulator has 3 apps, app1 app2 and app3
+            const appIds = ["app1", "app2", "app3"];
+
+            // readdir finds appIds
+            const mockReadDir = sinon.stub();
+            mockReadDir.withArgs(path.join(findSimulatorHomeResult, prefix)).returns(Q.resolve(appIds));
+            mockReadDir.throws();
+
+            // Only app2 has a plist file with thus bundle name
+            const existingPlistFile = path.join(findSimulatorHomeResult, prefix, "app2", suffix, `${bundleId}.plist`);
+
+            // existsSync only finds existingPlistFile to exist
+            const mockExistsSync = sinon.stub();
+            mockExistsSync.withArgs(existingPlistFile).returns(true);
+            mockExistsSync.returns(false);

```

Comment:
Yes. The code will check as many paths as there are entries in the directory (in this case we return `appIds` so one path for each of those) and we are saying that only one of those will be populated.